drivers/src/memory_layout.rs

```diff
@@ -22,24 +22,49 @@ use crate::PersistentData;
 pub const ROM_ORG: u32 = 0x00000000;
 pub const MBOX_ORG: u32 = 0x30040000;
 pub const ICCM_ORG: u32 = 0x40000000;
-pub const DCCM_ORG: u32 = 0x50000000;
-pub const ROM_DATA_ORG: u32 = 0x50000000;
-pub const CFI_STATE_ORG: u32 = 0x500003E4; // size = 6 words
-pub const BOOT_STATUS_ORG: u32 = 0x500003FC;
-pub const PERSISTENT_DATA_ORG: u32 = 0x50000400;
 
-pub const DATA_ORG: u32 = PERSISTENT_DATA_ORG + size_of::<PersistentData>() as u32;
+/// DCCM (Data Closely Coupled Memory)
+///
+/// Layout
+///
+/// | Partition       | Field        | Start       | End         | Size                 |
+/// |-----------------|--------------|-------------|-------------|----------------------|
+/// | DCCM            |              | 0x5000_0000 | 0x5004_0000 | 0x4_0000             |
+/// | Stack           |              | 0x5000_0000 | 0x5002_3800 | 0x2_3800             |
+/// |                 | FW Stack     | 0x5000_0000 | 0x5002_3000 | 0x2_3000             |
+/// |                 | ROM Stack    | 0x5000_0000 | 0x5001_2c00 | 0x1_2c00             |
+/// |                 | EStack       | 0x5002_3000 | 0x5002_3400 | 0x400                |
```

Comment:
I worry a little about keeping the memory_layout.rs code, this comment, and the `rom.ld` in sync moving forward.

Reply:
is it possible to autogenerate all of these somehow from a single source of truth?

Reply:
Is it okay, if I do this is a later PR?